src/lib.rs

```diff
@@ -1458,4 +1465,22 @@ mod test {
         assert_eq!(a[&2], "two");
         assert_eq!(a[&3], "three");
     }
+
+    // test serde trait implementations
```

Comment:
Maybe this is naive, but I wanted to make sure it actually compiled if it wasn't ever called. Does type checking happen before dead code is pruned?